activerecord/lib/active_record/connection_adapters/connection_specification.rb

```diff
@@ -126,42 +159,35 @@ def initialize(configurations)
         #
         # Full hash Configuration.
         #
-        #   configurations = { "production" => { "host" => "localhost", "database" => "foo", "adapter" => "sqlite3" } }
-        #   Resolver.new(configurations).resolve(:production)
+        #   configurations = { "primary" => { "host" => "localhost", "database" => "foo", "adapter" => "sqlite3" } }
+        #   Resolver.new(configurations).resolve(:primary)
         #   # => { "host" => "localhost", "database" => "foo", "adapter" => "sqlite3"}
         #
         # Initialized with URL configuration strings.
         #
-        #   configurations = { "production" => "postgresql://localhost/foo" }
-        #   Resolver.new(configurations).resolve(:production)
+        #   configurations = { "primary" => "postgresql://localhost/foo" }
+        #   Resolver.new(configurations).resolve(:primary)
         #   # => { "host" => "localhost", "database" => "foo", "adapter" => "postgresql" }
         #
         def resolve(config)
           if config
             resolve_connection config
-          elsif env = ActiveRecord::ConnectionHandling::RAILS_ENV.call
-            resolve_symbol_connection env.to_sym
           else
             raise AdapterNotSpecified
           end
         end
 
-        # Expands each key in @configurations hash into fully resolved hash
-        def resolve_all
-          config = configurations.dup
-          config.each do |key, value|
-            config[key] = resolve(value) if value
-          end
-          config
-        end
```

Comment:
This makes me wonder how much we need to worry about people calling all the available Hash methods on `ActiveRecord::Base.configurations[some_env]` and if we can just normalize the configurations up front.